marshal/writer_reflect.go

```diff
@@ -228,9 +228,9 @@ var wsL sync.Mutex
 
 func WriterFor(rt reflect.Type) (wr *TWriter) {
 	rtid := reflect.ValueOf(rt).Pointer()
+	wsL.Lock()
+	defer wsL.Unlock()
```

Comment:
Same here.

Either atomic.LoadPointer, or atomic.Value should be used.
Second is more preferred, i believe.

Or even sync.Map. It will be a bit slower on read, but cleanest choice.